server/readonly/readonly_service.ts

```diff
@@ -104,7 +105,12 @@ export class ReadonlyService extends BaseReadonlyService {
         return false;
       }
 
-      const authInfo = await this.securityClient.authinfo(request, headers);
+      const authType = cookie
+        ? cookie.isAnonymousAuth
+          ? AuthType.ANONYMOUS
+          : cookie.authType
+        : '';
```

Comment:
Shouldn't cookie always exist at this point? Can we clean this up instead of chaining ?'s? I got slightly confused reading this

Reply:
if you look at the code above, cookie is fetched via `const cookie = await this.securitySessionStorageFactory.asScoped(request).get();` which has a chance that it will return null if session storage was cleared. I think we can leave it as is.